### ISSUE TITLE
UseMySql missing option CharSetBehavior(CharSetBehavior.NeverAppend)

### ISSUE THREAD

**Issue by softevolution** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1435]
### Steps to reproduce

I used to add the db context in Startup.ConfigureServices() with code like this:

` 
services.AddDbContext<ApplicationDbContext>(
            dbContextOptions => dbContextOptions
                .UseMySql(Configuration.GetConnectionString("DefaultConnection"), 
                    new MariaDbServerVersion(new Version(10, 5)),
                    mySqlOptions => mySqlOptions
                        .CharSetBehavior(CharSetBehavior.NeverAppend)
        ));`

### The issue

I think that since the update to .NET 5 this is no longer possible. I get the compiler error:

'MySqlDbContextOptionsBuilder' does not contain a definition for 'CharSetBehavior' and no accessible extension method 'CharSetBehavior' accepting a first argument of type 'MySqlDbContextOptionsBuilder' could be found...

Can someone tell me how to achive the same effect as "CharSetBehavior(CharSetBehavior.NeverAppend)" previously had.

I've updated my project and now charset annotation get added which breaks compatibility with existing database columns.

### Further technical details

MySQL version: MariaDB 10.5
Operating system: Winsow 10
Pomelo.EntityFrameworkCore.MySql version: 5.0.0
Microsoft.AspNetCore.App version: 5.0

**Comment by mguinness** [comment]
CharSetBehavior was removed in https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/pull/1244.  Use `HasCharSet()` instead.

**Comment by lauxjpn** [comment]
The equivalence of the `CharSetBehavior.NeverAppend` option in Pomelo 5.0 in your case, is probably an explicit call to `HasCharSet()` on the model level:

```
modelBuilder.HasCharSet("utf8mb4", ApplyToDatabases);    // <-- if you don't mind an explicit charset for the database
modelBuilder.HasCharSet((string)null, ApplyToDatabases); // <-- if you do mind
```

The second option/line currently needs the explicit casting, because it would be ambiguous otherwise due to an additional overload that exists. It's an open bug that will be fixed shortly. Because of that, if you currently use migrations, the first line is probably what you want to use, unless you don't mind adding the explicit string-null-cast to newly generated migration files, until we fixed it.

**Comment by lauxjpn** [comment]
> The second option/line currently needs the explicit casting, because it would be ambiguous otherwise due to an additional overload that exists.

This has been fixed with #1448 and 1a1d8e95bbc29b575b22387b7e9c4fb1c83a2808.

---

@softevolution I'll close this one for now. If you think there is anything left to do for us here, please reopen it and explain to us, what still needs to be done.

**Comment by kikoron** [comment]
Please Change the [Configuration-Options Wiki](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/wiki/Configuration-Options)

**Comment by lauxjpn** [comment]
@kikoron Thanks, I updated the wiki for 5.0.

**Comment by lauxjpn** [comment]
@dcg90 What is the exact Pomelo version you are using?
Not specifying any explicit charset should be equivalent to `modelBuilder.HasCharSet("utf8mb4", DelegationModes.ApplyToAll);`.

**Comment by jonatanfelipesiqueira** [comment]
before there was no OldAnnotation and .Annotation in migrations, is there any way to continue the same as 3.1?

**Comment by jonatanfelipesiqueira** [comment]
> @dcg90 Qual é a versão exata do Pomelo que você está usando?
> Não especificar qualquer charset explícito deve ser equivalente a .`modelBuilder.HasCharSet("utf8mb4", DelegationModes.ApplyToAll);`

before there was no OldAnnotation and .Annotation in migrations, is there any way to continue the same as 3.1?

**Comment by lauxjpn** [comment]
> before there was no OldAnnotation and .Annotation in migrations, is there any way to continue the same as 3.1?

@jonatanfelipesiqueira I filed #1506 that I think is about what the issue in your case is likely to be. I already pushed PR to fix it (see #1507), which will be part of `5.0.2`.

**Comment by lauxjpn** [comment]
@dcg90 I am able to reproduce this and filed #1508.
